velox/functions/prestosql/registration/EnumFunctionsRegistration.cpp

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/functions/Registerer.h"
+#include "velox/functions/prestosql/EnumFunctions.h"
+#include "velox/functions/prestosql/types/BigintEnumRegistration.h"
+#include "velox/functions/prestosql/types/BigintEnumType.h"
+#include "velox/type/SimpleFunctionApi.h"
+
+namespace facebook::velox::functions {
+
+void registerEnumFunctions(const std::string& prefix) {
+  registerBigintEnumType();
+  std::vector<exec::SignatureVariable> constraints = {
```

Comment:
Why do we need this? Is E1 not enough?

Reply:
I was able to implement the `run` function for`TypeAnalysis<facebook::velox::BigintEnumT<E>>` , so that the SignatureVariable will be added through the `analysis` step using the template arguments of registerFunctions, rather than pass them in as constraints and have them added that way. Please let me know how this looks!